### PR TITLE
RFC: mark these methods as eligible for the faster cache for novel lookups

### PR DESCRIPTION
**Pull request by vtjnash**
Some observed methods that we may specialize often

### CODE REVIEW

File: base/dict.jl [hunk 1]
```diff
@@ -351,7 +351,7 @@ function ht_keyindex2!(h::Dict{K,V}, key) where V where K
     return ht_keyindex2!(h, key)
 end
 
-@propagate_inbounds function _setindex!(h::Dict, v, key, index)
+@propagate_inbounds @inline function _setindex!(h::Dict, v, key, index)
```
Comment (simeonschaub) [inline base/dict.jl:354]: Doesn't `@propagate_inbounds` already do inlining?

Comment (vtjnash) [reply base/dict.jl:354]:
```suggestion
@propagate_inbounds function _setindex!(h::Dict, v, key, index)
```
Indeed, that is what the docs say.

File: base/dict.jl [hunk 2]
```diff
@@ -392,6 +393,22 @@ function setindex!(h::Dict{K,V}, v0, key::K) where V where K
     return h
 end
 
+function setindex!(h::Dict{K,Any}, v, key::K) where K
```
Comment (oscardssmith) [inline base/dict.jl:396]: why not `setindex!(h::Dict{K,Any}, @nospecialize v, key::K)`?

Comment (vtjnash) [reply base/dict.jl:396]: I have options, so I use thehm

**Pull request merged**